File: algorithms/2_medium/M/02592_maximize_greatness_of_an_array/solution.cpp
class Solution {
public:
    static int maximizeGreatness(vector<int>& nums) {
        ranges::sort(nums);

        const size_t len = nums.size();
        size_t lo = 0;

        for (size_t hi = 0; hi < len; hi += 1) {
            if (nums[hi] > nums[lo]) {
                lo += 1;
            }
        }

        return static_cast<size_t>(lo);
    }
};

File: algorithms/2_medium/M/02592_maximize_greatness_of_an_array/solution.rs
impl Solution {
    pub fn maximize_greatness(mut nums: Vec<i32>) -> i32 {
        nums.sort_unstable();

        let len = nums.len();
        let mut lo: usize = 0;

        for hi in 0..len {
            if nums[hi] > nums[lo] {
                lo += 1;
            }
        }

        lo.try_into().unwrap()
    }
}
